sds/src/match_validation/http_validator.rs

```diff
@@ -18,13 +19,16 @@ pub struct HttpValidator {
 }
 
 impl HttpValidator {
-    pub fn new(config: HttpValidatorConfig) -> Self {
-        HttpValidator { config }
+    pub fn new_clone_config(config: &HttpValidatorConfig) -> Self {
+        HttpValidator {
+            config: config.clone(),
+        }
```

Comment:
Instead of cloning the class in the `new` method, I think it is more idiomatic to keep the keep the ownership of the config attribute in this class and to push the clone to the in the caller.